cmd/hub/mcscontroller/controller.go

```diff
@@ -118,15 +121,205 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 //	 automatically deleted by k8s when their ServiceExport is deleted.
 //
 func (r *ServiceExportReconciler) reconcile(ctx context.Context, name types.NamespacedName) (ctrl.Result, error) {
-	_ = r.Log.WithValues("service", name)
+	log := r.Log.WithValues("service", name)
+	//1:
+	var exportList mcsv1a1.ServiceExportList
+	if err := r.Client.List(ctx, &exportList, client.MatchingLabels{"name": name.Name, "namespace": name.Namespace}); err != nil {
+		log.Error(err, "unable to list service's service export")
+		return ctrl.Result{}, err
+	}
+	//2:
+	primaryEs, err := getPrimaryExportObject(exportList)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	//3:
+	for _, se := range exportList.Items {
+		if se.DeletionTimestamp != nil { //3a:
+			continue
+		}
 
-	//------------------------------------------
-	// -- @TODO CODE ABOVE SPECIFIED BEHAVIOR --
-	//------------------------------------------
+		exportSpec := &mcs.ExportSpec{}
+		err := exportSpec.UnmarshalObjectMeta(&se.ObjectMeta)
+		if err != nil {
+			log.Info("Failed to unmarshal exportSpec from serviceExport")
+			return ctrl.Result{}, err
+		}
 
+		err = exportSpec.IsCompatibleWith(primaryEs)
+		//3b:
+		if err != nil {
+			r.updateServiceExportConditions(ctx, &se, mcsv1a1.ServiceExportConflict, corev1.ConditionTrue, "", err.Error())
+			r.Client.Status().Update(ctx, &se)
+			//r.mcsClient.UpdateStatus(ctx, &se, metav1.UpdateOptions{})
+			err = r.deleteImport(ctx, &se)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+		} else { //3c:
+			r.updateServiceExportConditions(ctx, &se, mcsv1a1.ServiceExportValid, corev1.ConditionTrue, "", "")
+			r.Client.Status().Update(ctx, &se)
+			//se, err := r.mcsClient.UpdateStatus(ctx, &se, metav1.UpdateOptions{})
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			r.ensureImportFor(ctx, &se)
+		}
+	}
 	return ctrl.Result{}, nil
 }
 
+// getPrimaryExportObject finds the Primary ExportSpec from ServiceExportList.
+// Retruns the ExportSpec object that determined as the primary one from
+// the given ServiceExportList according to the conflict resolution specification set in the
+// KEP, that is implemented in IsPreferredOver method of ExportSpec.
+func getPrimaryExportObject(exportList mcsv1a1.ServiceExportList) (*mcs.ExportSpec, error) {
+	var primary *mcs.ExportSpec
+	for i, se := range exportList.Items {
+		exportSpec := &mcs.ExportSpec{}
+		err := exportSpec.UnmarshalObjectMeta(&se.ObjectMeta)
+		if err != nil {
+			return nil, err
+		}
+		if i == 0 || exportSpec.IsPreferredOver(primary) {
+			primary = exportSpec
+		}
+	}
+	return primary, nil
+}
+
+// create or update the ServiceImport corresponding to the provided export.
+func (r *ServiceExportReconciler) ensureImportFor(ctx context.Context, se *mcsv1a1.ServiceExport) error {
+	es := &mcs.ExportSpec{}
+
+	err := es.UnmarshalObjectMeta(&se.ObjectMeta)
+	if err != nil {
+		return err
+	}
+	namespacedName := types.NamespacedName{
```

Comment:
I think name should be `se.Name` and namespace should be `se.Namespace`. Then `es.Name` and `es.Namespace` should be stored in `lighthouse.submariner.io/sourceName` and `.../sourceNamespace`. In other words, the ServceImport should be stored in the submariner-k8s-broker namespace until a syncer imports it into a cluster.